src/System.Linq/src/System/Linq/Single.cs

```diff
@@ -149,5 +150,79 @@ public static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source,
 
             return default(TSource);
         }
+
+        public static bool TrySingle<TSource>(this IEnumerable<TSource> source, out TSource element)
+        {
+            if (source == null)
```

Comment:
This is correct for the reason given, and part of the motivation of the issue that asked for this API change seemed to be precisely this difference in how multiple elements are handled.